packages/validator/src/metrics.ts

```diff
@@ -0,0 +1,316 @@
+  inc(value?: number): void;
+  inc(labels: Labels, value?: number): void;
+  inc(arg1?: Labels | number, arg2?: number): void;
+
+  set(value: number): void;
+  set(labels: Labels, value: number): void;
+  set(arg1?: Labels | number, arg2?: number): void;
+
+  addCollect(collectFn: CollectFn<Labels>): void;
+}
+
+interface Histogram<Labels extends LabelsGeneric = never> {
+  startTimer(): () => number;
+
+  observe(value: number): void;
+  observe(labels: Labels, values: number): void;
+  observe(arg1: Labels | number, arg2?: number): void;
+
+  reset(): void;
+}
+
+interface AvgMinMax<Labels extends LabelsGeneric = never> {
+  set(values: number[]): void;
+  set(labels: Labels, values: number[]): void;
+  set(arg1?: Labels | number[], arg2?: number[]): void;
+}
+
+type GaugeConfig<Labels extends LabelsGeneric> = {
+  name: string;
+  help: string;
+  labelNames?: keyof Labels extends string ? (keyof Labels)[] : undefined;
+};
+
+type HistogramConfig<Labels extends LabelsGeneric> = {
+  name: string;
+  help: string;
+  labelNames?: (keyof Labels)[];
+  buckets?: number[];
+};
+
+type AvgMinMaxConfig<Labels extends LabelsGeneric> = GaugeConfig<Labels>;
+
+export interface MetricsRegister {
+  gauge<T extends LabelsGeneric>(config: GaugeConfig<T>): Gauge<T>;
+  histogram<T extends LabelsGeneric>(config: HistogramConfig<T>): Histogram<T>;
+  avgMinMax<T extends LabelsGeneric>(config: AvgMinMaxConfig<T>): AvgMinMax<T>;
+}
+
+export type Metrics = ReturnType<typeof getMetrics>;
+
+export type LodestarGitData = {
+  /** "0.16.0" */
+  semver: string;
+  /** "developer/feature-1" */
+  branch: string;
+  /** "4f816b16dfde718e2d74f95f2c8292596138c248" */
+  commit: string;
+  /** "0.16.0 developer/feature-1 ac99f2b5" */
+  version: string;
+  /** "prater" */
+  network: string;
+};
+
+/**
+ * A collection of metrics used throughout the Gossipsub behaviour.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function getMetrics(register: MetricsRegister, gitData: LodestarGitData) {
+  // Using function style instead of class to prevent having to re-declare all MetricsPrometheus types.
+
+  // Track version, same as https://github.com/ChainSafe/lodestar/blob/6df28de64f12ea90b341b219229a47c8a25c9343/packages/lodestar/src/metrics/metrics/lodestar.ts#L17
+  register
+    .gauge<LodestarGitData>({
+      name: "lodestar_version",
+      help: "Lodestar version",
+      labelNames: Object.keys(gitData) as (keyof LodestarGitData)[],
+    })
+    .set(gitData, 1);
+
+  return {
+    // Attestation journey:
+    // - Wait for block or 1/3, call prepare attestation
+    // - Get attestation, sign, call publish
+    // - Wait 2/3, call prepare aggregate
+    // - Get aggregate, sign, call publish
+
+    // Max wait time is 1 / 3 of slot = 12 / 3 = 4 sec
+    attesterStepCallProduceAttestation: register.histogram({
```

Comment:
right now there is I/O lag issue at the beacon node side in contabo nodes - feature group, so I'd like to track the time before we produce and right after we produce attestation.

Reply:
That is tracked by the REST API client metrics